data_processing/pathology/proxy_table/regional_annotation/generate.py

```diff
@@ -0,0 +1,298 @@
+    if os.path.exists(zipfile_path):
+        os.remove(zipfile_path)
+
+    return row
+
+def create_proxy_table():
+    '''
+    Creates the pathology annotations proxy table with information contained in the specified data_config_file
+
+    :param data_config_file: data configuration
+    :param app_config_file: app configuration
+    :return: exit_code = 0 if successful, > 0 if unsuccessful
+    '''
+    # pd.set_option("display.max_rows", None, "display.max_columns", None)
+    # pd.set_option('display.max_colwidth', -1)
+
+    exit_code = 0
+    cfg = ConfigSet()
+    spark = SparkConfig().spark_session(
+                                 config_name=const.APP_CFG,
+                                 app_name="data_processing.pathology.proxy_table.annotation.generate")
+
+    SLIDEVIEWER_API_URL = cfg.get_value(path=const.DATA_CFG + '::SLIDEVIEWER_API_URL')
+    SLIDEVIEWER_CSV_FILE = cfg.get_value(path=const.DATA_CFG + '::SLIDEVIEWER_CSV_FILE')
+    PROJECT_ID = cfg.get_value(path=const.DATA_CFG + '::PROJECT_ID')
+    LANDING_PATH = cfg.get_value(path=const.DATA_CFG + '::LANDING_PATH')
+    slides = fetch_slide_ids(SLIDEVIEWER_API_URL, PROJECT_ID, LANDING_PATH, SLIDEVIEWER_CSV_FILE)
+
+    schema = StructType([
+        StructField('slideviewer_path', StringType()),
+        StructField('slide_id', StringType()),])
+
+    df = spark.createDataFrame(slides, schema)
+
+    # populate columns
+    TMP_ZIP_DIR = cfg.get_value(const.DATA_CFG + '::REQUESTOR_DEPARTMENT') + '_tmp_zips'
+    df = df.withColumn('sv_project_id', lit(cfg.get_value(const.DATA_CFG+'::PROJECT_ID'))) \
+        .withColumn('bmp_filepath', lit('')) \
+        .withColumn('users', array([lit(user) for user in cfg.get_value(const.DATA_CFG + '::USERS')])) \
+        .withColumn('date_added', current_timestamp()) \
+        .withColumn('date_updated', current_timestamp()) \
+        .withColumn('bmp_record_uuid', lit('')) \
+        .withColumn('latest', lit(True)) \
+        .withColumn('SLIDE_BMP_DIR', lit(os.path.join(cfg.get_value(const.DATA_CFG+'::LANDING_PATH'), 'regional_bmps'))) \
+        .withColumn('TMP_ZIP_DIR', lit(os.path.join(cfg.get_value(const.DATA_CFG+'::LANDING_PATH'), TMP_ZIP_DIR))) \
+        .withColumn('SLIDEVIEWER_API_URL', lit(cfg.get_value(const.DATA_CFG + '::SLIDEVIEWER_API_URL'))) \
+
+    # explore by user list
+    df = df.select('slideviewer_path',
+                   'slide_id',
+                   'sv_project_id',
+                   'bmp_filepath',
+                   explode('users').alias('user'),
+                   'date_added',
+                   'date_updated',
+                   'bmp_record_uuid',
+                   'latest',
+                   'SLIDE_BMP_DIR',
+                   'TMP_ZIP_DIR',
+                   'SLIDEVIEWER_API_URL'
+                   )
+
+    df = df.groupby(['slideviewer_path', 'user']) \
+        .applyInPandas(process_regional_annotation_slide_row_pandas, schema=df.schema)
+
+    df = df.toPandas()
```

Comment:
Line 195-212 could be rewritten using spark dataframe, to avoid switching back and forth between pandas/spark dataframes.

Reply:
Yes. I'll revisit next week with a fresh mind and try to refine code in this package further.